### PR TITLE
convert landing page & module page to typescript

### PR DESCRIPTION
**Pull request by maxpinheiro**
I'm not the most familiar with TS so please lmk if my work is messy

### CODE REVIEW

File: package.json
Comment (krisharma) [inline package.json:10]: @jakeisnt from my understanding we should avoid having unpinned dependencies?

File: yarn.lock
Comment (krisharma) [inline yarn.lock:1800]: @jakeisnt another note of the unpinned dependency - this would go away on rerunning `yarn install` but noting again in case

**Review by krisharma — CHANGES_REQUESTED** [review]
@jakeisnt I left a comment about the package pinning but please also check the rest of the PR - thanks!

File: package.json
Comment (jakeisnt) [reply package.json:10]: yeah, it's not a huge deal with some packages, but as i mentioned before i noticed that the tailwindcss styles changed at some point last semester because the dependency wasn't fixed to a specific version, and having the package version change on `yarn install` could result in similar issues in the future

Comment (jakeisnt) [reply package.json:10]: even though it is not a big deal this should still be pinned

Comment (maxpinheiro) [reply package.json]: just pinned the dependency

**Comment by jakeisnt** [comment]
Instead of leaving messages like 'merge pr', 'merge' and 'merge conflict' in the git history, it's best practice to either rename the commits or combine them with other commits in the history before merging them with the master branch so that the history of the codebase is easy to reference in the future.

You can do this by using the terminal command `git rebase -i <commit hash>`, where the commit hash is the long string of characters visible when viewing a specific commit (i.e. [clicking here](https://github.com/sandboxnu/a-eye/pull/91/commits/d24097d82f86682dbb4159aa8cca72745e8dbf6d)). This runs a rebase on all of the commits between your current status and the commit associated with the commit hash, and provides a menu allowing you to change commit messages, merge multiple commits together and more.

Unfortunately it looks like there is no way to do this with vscode by default (you have to install [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens#interactive-rebase-editor-) and learn how to use it), but I found [this reference for github desktop](https://github.com/desktop/desktop/blob/development/docs/technical/rebase-flow.md). Happy to help with this!

File: src/landingPage/ModuleIntro.tsx
Comment (jakeisnt) [inline src/landingPage/ModuleIntro.tsx:3]: It would be helpful to add some documentation here for the props. I found [this file](https://github.com/sandboxnu/a-eye/blob/master/src/aboutPage/ProfileCard.tsx) that has an example and [this looks to be a decent tutorial for adding JSDoc to react components](https://www.inkoop.io/blog/a-guide-to-js-docs-for-react-js/). Whether this is necessary is ultimately up to @krisharma though!

File: src/modulePage/ModulePage.tsx
Comment (jakeisnt) [inline src/modulePage/ModulePage.tsx:9]: Strings for the keys aren't necessary in Typescript definitions (or Javascript objects), so you can remove them here if you'd like. (They are necessary for JSON, which can get a bit confusing)

File: src/modulePage/ModulePage.tsx
Comment (jakeisnt) [inline src/modulePage/ModulePage.tsx:42]: Good fix!

File: src/modulePage/ModuleSection.tsx
Comment (jakeisnt) [inline src/modulePage/ModuleSection.tsx:79]: If 'labelColorHex' isn't used, I'd define a default prop in the `PCADemo` component or omit it from that component entirely.

**Review by jakeisnt — COMMENTED** [review]
Code looks good! Left a couple of minor comments. I see the commit organization as a big deal but I'll yield to @krisharma 's opinion on that.

File: src/landingPage/ModuleIntro.tsx
Comment (jakeisnt) [reply src/landingPage/ModuleIntro.tsx:3]: I'd go through and do this for the other components you've defined if we want to add the documentation

Comment (krisharma) [reply src/landingPage/ModuleIntro.tsx:3]: Yes JSDoc would be great! Just good practice for us moving forward - thanks for catching this Jake

File: src/modulePage/ModuleSection.tsx
Comment (krisharma) [reply src/modulePage/ModuleSection.tsx:79]: Nice, didn't know you could do this

**Comment by maxpinheiro** [comment]
Krish and I decided to leave the commit organization as-is for this PR, but we'll make sure to switch to the better form from now on.

**Review by jakeisnt — APPROVED** [review]
lgtm

**Review by krisharma — APPROVED** [review]
LGTM, go for it!

**Pull request merged**
